pkg/monitor/intervalcreation/node.go

```diff
@@ -82,32 +62,32 @@ func IntervalsFromEvents_NodeChanges(events monitorapi.Intervals, _ monitorapi.R
 		// excessively long intervals if the events are missing (because openInterval does not create
 		// a new interval).
 		switch reason {
+		case "NotReady":
+			nodeStateTracker.openInterval(nodeLocator, notReadyState, event.From)
+		case "Ready":
+			message := monitorapi.Message().Reason(monitorapi.NodeNotReadyReason).Messagef("role/%v is not ready", roles)
+			intervals = append(intervals, nodeStateTracker.closeInterval(nodeLocator, notReadyState, simpleCondition(monitorapi.Warning, monitorapi.NodeNotReadyReason, message), event.From)...)
 		case "MachineConfigChange":
-			openInterval(state, "Update", event.From)
+			nodeStateTracker.openInterval(nodeLocator, updateState, event.From)
 		case "MachineConfigReached":
-			message := strings.ReplaceAll(event.Message, "reason/MachineConfigReached ", "reason/NodeUpdate phase/Update ") + " roles/" + roles
-			closeInterval(state, "Update", monitorapi.Info, event.Locator, message, event.From)
+			message := strings.ReplaceAll(event.Message, "reason/MachineConfigReached ", "phase/Update ") + " roles/" + roles
+			intervals = append(intervals, nodeStateTracker.closeInterval(nodeLocator, updateState, simpleCondition(monitorapi.Info, monitorapi.NodeUpdateReason, message), event.From)...)
 		case "Cordon", "Drain":
-			openInterval(state, "Drain", event.From)
+			nodeStateTracker.openInterval(nodeLocator, drainState, event.From)
 		case "OSUpdateStarted":
-			closeInterval(state, "Drain", monitorapi.Info, event.Locator, fmt.Sprintf(msgPhaseDrain, roles), event.From)
-			openInterval(state, "OperatingSystemUpdate", event.From)
+			intervals = append(intervals, nodeStateTracker.closeInterval(nodeLocator, drainState, simpleCondition(monitorapi.Info, monitorapi.NodeUpdateReason, fmt.Sprintf(msgPhaseDrain, roles)), event.From)...)
+			nodeStateTracker.openInterval(nodeLocator, osUpdateState, event.From)
 		case "Reboot":
-			closeInterval(state, "Drain", monitorapi.Info, event.Locator, fmt.Sprintf(msgPhaseDrain, roles), event.From)
-			closeInterval(state, "OperatingSystemUpdate", monitorapi.Info, event.Locator, fmt.Sprintf(msgPhaseOSUpdate, roles), event.From)
-			openInterval(state, "Reboot", event.From)
+			intervals = append(intervals, nodeStateTracker.closeInterval(nodeLocator, drainState, simpleCondition(monitorapi.Info, monitorapi.NodeUpdateReason, fmt.Sprintf(msgPhaseDrain, roles)), event.From)...)
+			intervals = append(intervals, nodeStateTracker.closeInterval(nodeLocator, osUpdateState, simpleCondition(monitorapi.Info, monitorapi.NodeUpdateReason, fmt.Sprintf(msgPhaseOSUpdate, roles)), event.From)...)
+			nodeStateTracker.openInterval(nodeLocator, rebootState, event.From)
```

Comment:
Is it necessary to have this embedded logic about one reason implying closing intervals for other reasons? Any changes in the product and we're immediately broken. Could we instead just close any matching open interval?